pkg/shell/cockpit-dashboard.js

```diff
@@ -410,7 +410,18 @@ PageDashboard.prototype = {
                 update_series();
             }
 
-            return render;
+            function throttled_render() {
+                var timer = null;
+                return function () {
+                    if (timer)
+                        window.clearTimeout(timer);
```

Comment:
If I call throttled_render every x ms (x<500), render() will be delayed forever.

Easiest change: if there already is a timer, return directly. Downside: renders will be delayed even if they occur rarely. Advanced mechanics if we can expect that to happen: Save time of last render t0 in each render call. If time of current call t1 is in "quiet" period shorter than the limit L, i.e. (t1-t0)<L, create a render timeout for L-(t1-t0), unless a timer already exists. If t1-t0>=L, call render directly.